components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/par/OAuth2ParEndpoint.java

```diff
@@ -235,14 +238,56 @@ private void validateAuthzRequest(HttpServletRequest request) throws ParCoreExce
 
         try {
             OAuthAuthzRequest oAuthAuthzRequest = getOAuthAuthzRequest(request);
-            validateRequestObject(oAuthAuthzRequest);
+            RequestObject requestObject = validateRequestObject(oAuthAuthzRequest);
+            Map<String, String> oauthParams = overrideRequestObjectParams(request, requestObject);
+            if (isFAPIConformantApp(oAuthAuthzRequest.getClientId())) {
+                EndpointUtil.validateFAPIResponseMode(oauthParams.get(RESPONSE_TYPE), oauthParams.get(RESPONSE_MODE));
```

Comment:
lets change the method name in the util to give more clarity. Current method name reads like that there is a response mode named "FAPI"
```suggestion
                EndpointUtil.validateFAPIAllowedResponseMode(oauthParams.get(RESPONSE_TYPE), oauthParams.get(RESPONSE_MODE));
```